llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -313,22 +323,36 @@ void validateUsedAspectsForFunctions(const FunctionToAspectsMapTy &Map,
       continue;
 
     Function *F = It.first;
-
-    // Entry points will have their declared aspects from their kernel call.
-    // To avoid double warnings, we skip them.
-    if (std::find(EntryPoints.begin(), EntryPoints.end(), F) !=
-        EntryPoints.end())
-      continue;
-
-    const MDNode *DeviceHasMD = F->getMetadata("sycl_declared_aspects");
-    if (!DeviceHasMD)
-      continue;
+    bool FIsEntryPoint = std::find(EntryPoints.begin(), EntryPoints.end(), F) !=
+                         EntryPoints.end();
```

Comment:
```suggestion
    bool FIsEntryPoint = is_contained(EntryPoints, F);
```

By using `llvm::is_contained` from `ADT/STLExtras.h`. I think we can even inline this shorter version without loosing readability.

Reply:
Ah, that's a useful utility function! I have inlined it as suggested. Thanks!